pkg/registry/core/serviceaccount/strategy.go

```diff
@@ -61,7 +73,11 @@ func (strategy) AllowCreateOnUpdate() bool {
 }
 
 func (strategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
-	cleanSecretReferences(obj.(*api.ServiceAccount))
+	serviceAccount := obj.(*api.ServiceAccount)
+	cleanSecretReferences(serviceAccount)
+	if hasEnforceMountableSecretsAnnotation(obj.(*api.ServiceAccount)) {
+		warning.AddWarning(ctx, "", fmt.Sprintf("deprecated annotation %q on service account %s/%s", EnforceMountableSecretsAnnotation, serviceAccount.Namespace, serviceAccount.Name))
+	}
```

Comment:
Probably easier to put all this into one `warnIfHasEnforceMountableSecretsAnnotation` function.